src/software/ai/hl/stp/action/intercept_ball_action.cpp

```diff
@@ -35,146 +36,132 @@ void InterceptBallAction::accept(MutableActionVisitor& visitor)
-             (ball_not_moving || ball_time_to_position > robot_time_to_pos)))
+        if (robot_time_to_pos < ball_time_to_position)
         {
-            intercept_pos = closest_point;
+            break;
         }
+        intercept_position += ball.velocity().normalize(0.1);
+    }
 
-        auto point_ball_leaves_field = getPointBallLeavesField(field, ball);
-        if (intercept_pos)
-        {
-            while (point_in_front_of_ball)
-            {
-                moveToInterceptPosition(yield, closest_point);
-
-                closest_point = closestPointOnLine(
-                    robot->position(),
-                    Line(ball.position(), ball.position() + ball.velocity()));
-                point_in_front_of_ball =
-                    acuteAngle(ball.velocity(), closest_point - ball.position()) <
-                    Angle::quarter();
-            }
-        }
-        else if (point_ball_leaves_field)
+    // Move to the intercept position and stop while avoiding the ball
+    bool robot_in_intercept_position = false;
+    do
+    {
+        yield(std::make_unique<MoveIntent>(robot->id(), intercept_position,
+                                           (-ball.velocity()).orientation(), 0, 0,
+                                           DribblerEnable::ON, MoveType::NORMAL,
+                                           AutokickType::NONE, BallCollisionType::AVOID));
+
+        if (ball.velocity().length() < BALL_MOVING_SLOW_SPEED_THRESHOLD / 2.0)
         {
-            LOG(DEBUG) << "ball leaving field" << std::endl;
-            yield(std::make_unique<MoveIntent>(
-                robot->id(), point_ball_leaves_field.value(),
-                (ball.position() - robot->position()).orientation(), 0, 0,
-                DribblerEnable::ON, MoveType::NORMAL, AutokickType::NONE,
-                BallCollisionType::ALLOW));
+            restart();
         }
-        else
+
+        bool intercept_point_in_front_of_ball =
+            acuteAngle(ball.velocity(), intercept_position - ball.position()) <
+            Angle::quarter();
+        bool robot_at_intercept_point =
+            distance(robot->position(), intercept_position) < 0.015;
+        bool robot_stopped = robot->velocity().length() < 0.02;
+        robot_in_intercept_position =
+            intercept_point_in_front_of_ball && robot_at_intercept_point && robot_stopped;
+    } while (!robot_in_intercept_position);
+
+    // Wait for the ball to arrive at the robot
+    bool intercept_done = false;
+    do
+    {
+        yield(std::make_unique<MoveIntent>(robot->id(), intercept_position,
+                                           (-ball.velocity()).orientation(), 0, 0,
+                                           DribblerEnable::ON, MoveType::NORMAL,
+                                           AutokickType::NONE, BallCollisionType::ALLOW));
+
+        if (!intercept_done &&
+            ball.velocity().length() < BALL_MOVING_SLOW_SPEED_THRESHOLD / 2.0)
```

Comment:
same as above for BALL_MOVING_SLOW_SPEED_THRESHOLD

Reply:
see above